src/transformers/data/data_collator.py

```diff
@@ -305,20 +305,30 @@ def torch_call(self, features):
 
         label_name = "label" if "label" in features[0].keys() else "labels"
         labels = [feature[label_name] for feature in features] if label_name in features[0].keys() else None
+
+        if labels is not None:
+            no_label_features = [{k: v for k, v in feature.items() if k != label_name} for feature in features]
+        else:
+            no_label_features = features
```

Comment:
I don't think you need the test here,
```suggestion
        no_label_features = [{k: v for k, v in feature.items() if k != label_name} for feature in features]
```
will always work.